packages/ruleset/src/functions/path-segment-case-convention.js

```diff
@@ -0,0 +1,37 @@
+const { casing } = require('@stoplight/spectral-functions');
+let casingConfig;
+
+module.exports = function(operation, options, { path }) {
+  // Save this rule's "functionOptions" value since we need
+  // to pass it on to Spectral's "casing" function.
+  casingConfig = options;
+
+  return pathSegmentCaseConvention(operation, path);
+};
+
+function pathSegmentCaseConvention(pathItem, path) {
+  // The path string (e.g. '/v1/resources/{resource_id}') will be the last element in 'path'.
+  const pathStr = path[path.length - 1];
+
+  // Parse the path string into the individual path segments
+  let pathSegments = pathStr.split('/');
+
+  // Filter out the path segments that are either "" (the first one due to the
+  // path string starting with "/") or a path param reference,
+  // since we don't want to raise errors for those.
+  pathSegments = pathSegments.filter(s => s !== '' && !s.match(/^{.*}$/));
+
+  const errors = [];
+  for (const segment of pathSegments) {
+    const result = casing(segment, casingConfig);
+    if (result) {
+      // Update the message to clarify that it is an operationId value that is incorrect,
+      // and update the path.
+      result[0].message = 'Path segments ' + result[0].message + ': ' + segment;
```

Comment:
I coded this to append the segment to the error message to avoid the situation where spectral was returning just a single error even if there were multiple incorrect path segments.
Apparently spectral merges together errors that have a common error message and path array, but I thought it would be good to return an error for each incorrect path segment, similar to the legacy rule implementation.